source/globalCommands.py

```diff
@@ -4280,16 +4280,19 @@ def script_reportLinkDestination(
 		positioned on a link, or an element with an included link such as a graphic.
 		:param forceBrowseable: skips the press once check, and displays the browseableMessage version.
 		"""
+		focus = api.getFocusObject()
 		try:
 			ti: textInfos.TextInfo = api.getCaretPosition()
+			link = ti._getLinkDataAtCaretPosition()
 		except RuntimeError:
-			log.debugWarning("Unable to get the caret position.", exc_info=True)
-			ti: textInfos.TextInfo = api.getFocusObject().makeTextInfo(textInfos.POSITION_FIRST)
-		link = ti._getLinkDataAtCaretPosition()
+			try:
+				link = focus.linkData
+			except NotImplementedError:
+				link = None
 		presses = scriptHandler.getLastScriptRepeatCount()
 		if link:
 			if link.destination is None:
-				# Translators: Informs the user that the link has no destination
+				# Translators: Reported when using the command to report the destination of a link.
 				ui.message(_("Link has no apparent destination"))
```

Comment:
Apparent seems strange here. Would this change be misleading?

```suggestion
				ui.message(_("Link has no destination"))
```

Reply:
This text was introduced in #14723 by @lukaszgo1 on a suggestion of @XLTechie (https://github.com/nvaccess/nvda/pull/14723#issuecomment-1480802046).

@CyrilleB79 what is your opinion? Maybe a rewording would be needed.

But from what I understand, the link may have a destination; it's just that we are not able to determine it.